Random tree games

iosonofabio opened the pull request:
Addressing  #343.

This raises an interesting point: I now made two new global variables in the igraph namespace for this: `RANDOM_TREE_PRUFER` and `RANDOM_TREE_LERW`: probably that's not a great idea? Perhaps we should check strings as parameters for the Python function instead?

e.g. 

```python

igraph.Graph.Random_Tree(10, False, igraph.RANDOM_TREE_PRUFER) # current
igraph.Graph.Random_Tree(10, False, "Prufer") # perhaps better?
```

That applies to a number of other situations, of course, so I'm wondering what is the better choice, see also #386

ntamas commented:
The way it typically works is that you add a conversion function in `src/convert.c` that converts from strings to the appropriate enum members, and then you use those enum members instead. I started moving away from global integer variables a while ago, although of course I cannot remove any existing stuff in order not to break backwards compatibility.

ntamas commented on src/_igraph/graphobject.c:
```diff
@@ -3400,6 +3400,60 @@ PyObject *igraphmodule_Graph_Tree(PyTypeObject * type,
+  static char *kwlist[] = { "n", "directed", "method", NULL };
+
+  if (!PyArg_ParseTupleAndKeywords(args, kwds, "lll", kwlist,
+                                   &n, &directed_o, &tree_method_o))
```
`directed` is a boolean so I'd rather go for `PyObject* directed_o` and use `PyObject_IsTrue(...)` for sake of consistency with how it's done elsewhere in the library.

As for the `method` argument, look into `convert.c` to see how it's typically done for other enums.

iosonofabio replied:
alright let me take a look

iosonofabio replied:
yep I've changed this now.

ntamas reviewed (changes requested):
Thanks a lot for the contribution! It looks good to me in general, except the handling of arguments, see my comments there.

iosonofabio commented:
I rebased on top of the realize_degree_sequence commits, changed this to use the strict-ish conversion function, and all tests pass, so I think it's ready for merging.

ntamas commented:
Thanks! I'll add default argument handling and then it's good to go.

szhorvat commented:
It'd be nice to have consistent capitalization for method names, i.e. `Prufer`, `LERW` or `prufer`, `lerw`. Now it's mixed.

I recommend Prufer as default unless the graph is directed. There is no difference in the distribution with the two methods, only in the performance.

ntamas commented:
Matching is case-insensitive, the docstrings consistently use lowercase everywhere, the tests use uppercase here and there to test the cast insensitiveness.

I used LERW as the default because then I don't need to make the default dependent on another variable. How big is the performance difference? If it is significant, I'll change the default.

szhorvat commented:
Would have to tes, but I think not significant.
